concent_api/core/payments/bankster.py

```diff
@@ -474,3 +477,18 @@ def discard_claim(deposit_claim: DepositClaim) -> bool:
         subtask_id=deposit_claim.subtask_id,
     )
     return claim_removed
+
+
+def validate_list_of_transaction_timestamp(
+        list_of_transactions: List[BatchTransferEvent],
+        acceptances: List[SubtaskResultsAccepted],
+) -> None:
+    """ Validate if cut_off_time < T2 + PDT, unless requestor already made some payment. """
+
+    cut_off_time = get_current_utc_timestamp()
+
+    youngest_payments_ts = max(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
+
+    if not any(youngest_payments_ts <= closure_time for closure_time in list_of_transactions):
```

Comment:
Those two conditions should be combined into single.

Reply:
This line change because of [this comment](https://github.com/golemfactory/concent/pull/1015#discussion_r241346283)

Reply:
> They can, but it made more logical sense to me like that, because we don't want to touch the second if, when first isn't True.

This is how Python works, if you have two conditions in `if` connected with `and` and first fails, second is not being run.

Reply:
> I have meant the first condition, should it really raise that exception and stop processing if the list is empty?

I think there's some misunderstanding. First condition doesn't raise any exception. If `list_of_transactions` is empty, we still need to check if `youngest_payments_ts` is not too young. And we raise exception only when it is.

Reply:
> I think there's some misunderstanding. First condition doesn't raise any exception. If list_of_transactions is empty, we still need to check if youngest_payments_ts is not too young. And we raise exception only when it is.

Makes sense, but it still should be combined into single `if`.

Reply:
Done